package/MDAnalysis/core/universe.py

```diff
@@ -113,20 +112,38 @@ def _topology_from_file_like(topology_file, topology_format=None,
         # one because the file isn't present
         # or the permissions are bad, second when the parser fails
         if (err.errno is not None and
-            errno.errorcode[err.errno] in ['ENOENT', 'EACCES']):
-            # Runs if the error is propagated due to no permission / file not found
+                errno.errorcode[err.errno] in ['ENOENT', 'EACCES']):
+            # Runs if the error is propagated due to no permission / file not
+            # found
             raise sys.exc_info()[1] from err
         else:
             # Runs when the parser fails
             raise IOError("Failed to load from the topology file {0}"
-                            " with parser {1}.\n"
-                            "Error: {2}".format(topology_file, parser, err))
+                          " with parser {1}.\n"
+                          "Error: {2}".format(topology_file, parser, err))
     except (ValueError, NotImplementedError) as err:
         raise ValueError(
             "Failed to construct topology from file {0}"
             " with parser {1}.\n"
             "Error: {2}".format(topology_file, parser, err))
-    return topology
+    txyz = False
```

Comment:
What is this section about? I do not like to have code mentioning specific formats in universe.py. The universes are supposed to be format agnostic so that somebody can write a new parser and have all the features without changing anything but their own parser code.

Reply:
the txyz parser is the only one that gues masses from names not atomtypes like the majority of other parsers, that's why I put this flag to handle this tiny deviation. However, when the automatic guessing of types and masses will be removed, lots of code pieces that seem redundant can be omitted safely. I just put it as a transition phase to make sure the old behavior of all parsers is still the same

Reply:
A more general, long-term solution could just be to pass the format/trajectory reader to the guesser. There might be other formats that differ in the future.

Reply:
From discussing, one way around this is since a Guesser has a reference to the Universe, the Guesser can see what format was initially supplied.